### DISCUSSION TITLE
How to Edit/add new properties

### DISCUSSION THREAD

**Discussion by Akshay3241** [IfcOpenShell/IfcOpenShell#3552] [Q&A]
I am trying to add new properties under predefined property set in IFC file using IfcOpenShell.
I am new to IFC and Python.
Anyone knows how to do that?

**Comment by Moult** [comment]
https://blenderbim.org/docs-python/autoapi/ifcopenshell/api/pset/edit_pset/index.html

**Reply by Akshay3241** [reply]
@Moult  Thank you very much

**Reply by Akshay3241** [reply]
I want to add new attributes into a IFC file. Can you help me into that @Moult

**Reply by Moult** [reply]
https://blenderbim.org/docs-python/ifcopenshell-python/hello_world.html#core-functionality-crash-course

**Reply by Akshay3241** [reply]
@Moult 
So, like Name or GlobalId can I add here wall.Color/type/position etc and assign its value?

**Reply by Moult** [reply]
Yep. `wall.Name = 'foo'`

**Reply by Akshay3241** [reply]
Hello @Moult 
I am trying to add reference in my IFC file. As per IFCOpenShell documentation I am tring to add reference for "IfcDoorType" but it is somehow not working. I tried the same for Door occurance i.e by "IfcDoor"[0] and it works. But I want it for all doors and want to execute for "IfcDoorType".

**Reply by Moult** [reply]
Can you share your code?

**Reply by Akshay3241** [reply]
door = model.by_type("IfcDoor")[0]
classification = ifcopenshell.api.run("classification.add_classification",  model, classification="Uniclass")
ifcopenshell.api.run("classification.add_reference", model,product=door, classification=classification, identification="Schindler_3241", name="")
model.write("C:\\Users\\sabaleak\\Downloads\\UniClass.ifc")

Here I want to add reference I am doing this for door occurance.
Instead of this I want to execute for "IfcDoorType" but facing error.
door_type = model.by_type("IfcDoorType")[0] 

This is also mention in (https://blenderbim.org/docs-python/autoapi/ifcopenshell/api/classification/add_reference/index.html) but not working and that's why I am forcefully doing this for single door occurance.

**Reply by Akshay3241** [reply]
[ExampleForIFCOpenShel_SDE_VDIl.zip](https://github.com/IfcOpenShell/IfcOpenShell/files/12368989/ExampleForIFCOpenShel_SDE_VDIl.zip)
I am using above IFC files as a model in code.

**Reply by Akshay3241** [reply]
Hello @Moult,

Do you need further information regarding my question. Please let me know,

Thanks in advance.

**Reply by Moult** [reply]
The issue is that you actually have no door types in your model. Your door occurrences have no relating type, and there are no types at all in your model. This isn't a bug in IfcOpenShell, it's just how your model data is structured.

**Reply by Akshay3241** [reply]
@Moult 
thank you for information.
Can u guide me here how to proceed?

**Comment by Akshay3241** [comment]
How to add/edit attributes to a IFC file?
